### PR TITLE
Issue #1620: Track Clicks on YouTube Tile

### PR DESCRIPTION
**Pull request by dnarcese**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
Just a telemetry change - testing telemetry is not set up yet
- [x] This PR includes a **CHANGELOG entry** or does not need one

### CODE REVIEW

**Comment by dnarcese** [comment]
# Request for data collection review form

1) What questions will you answer with this data?
- How many users & sessions use Firefox for Fire TV for things outside of YouTube?

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
* Determine the count & percent of sessions that include non-YouTube tile clicks.
* Determine the count & percent of users that click non-YouTube tiles.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
- This is the most accurate way to determine this information.

4) Can current instrumentation answer these questions?
- No telemetry currently exists to answer these questions.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.   

| Measurement Description                                      | Data Collection Category | Tracking Bug #                                               |
| ------------------------------------------------------------ | ------------------------ | ------------------------------------------------------------ |
| When the YouTube tile is clicked                             | 2                        | [Issue 1620](https://github.com/mozilla-mobile/firefox-tv/issues/1620) |
| How many sessions do not contain a click on the YouTube tile | 2                        | [Issue 1620](https://github.com/mozilla-mobile/firefox-tv/issues/1620) |

6) How long will this data be collected?
* This data will be permanently monitored by @athomasmoz

7) What populations will you measure?
* All users with telemetry enabled, on release channel.

8) If this data collection is default on, what is the opt-out mechanism for users?
- There is an option in settings to stop sending usage data.

9) Please provide a general description of how you will analyze this data.
- The data will be analyzed through Redash queries and dashboards.

10) Where do you intend to share the results of your analysis?
- On Redash and with mobile teams.

File: docs/telemetry.md [hunk 1]
```diff
@@ -54,20 +54,17 @@ The event ping contains a list of events ([see event format on readthedocs.io](h
...
-| Unique tiles clicked per session**     | aggregate| click                 | home_tile  | `<int>` | |
+| Unique tiles clicked per session     | aggregate| click                 | home_tile  | `<int>` | |
+| Sessions that **do not** contain a click on  YouTube tile | aggregate | click | home_tile | session_without_youtube | |
 | Menu shown by user              | action   | user_show       | menu         | ||
 | Menu hidden by user              | action   | user_hide       | menu         | ||
 | No menu action taken              | aggregate   | no_action_taken       | menu         | ||
 
 (*)When the pin site switch is clicked, the state (on/off) of the desktop mode switch is also sent.
 
-(**)This is broken: see [#879](https://github.com/mozilla-mobile/firefox-tv/issues/879).** For overlay hidden, we only capture cases where the user closes the overlay themselves:
```
Comment (liuche) [inline docs/telemetry.md]: I actually think this `**` is actually supposed to be on the "Menu hidden by user" telemetry probe, so don't remove it. The formatting of the `(**)` is also broken, I think.

File: app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetryIntegration.kt [hunk 2]
```diff
@@ -92,6 +92,7 @@ open class TelemetryIntegration protected constructor(
         val TILE_CUSTOM = "custom"
         val POCKET_VIDEO_MEGATILE = "pocket_video_tile"
         val YOUTUBE_TILE = "youtube_tile"
+        val SESSION_WITHOUT_YOUTUBE = "session_without_youtube"
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetryIntegration.kt]: I talked to Su (who is doing metrics for this) and I think that this doesn't need to be collected - he said it's easy enough to group by sessions that don't have a youtube_tile event, so can we remove this commit? (and the telemetry docs for it)

File: docs/telemetry.md (hunk 1 above)
Comment (dnarcese) [reply docs/telemetry.md]: Issue #879 is closed, so I figured this isn't actually broken anymore- which is why I removed it.

File: docs/telemetry.md [hunk 3]
```diff
@@ -54,20 +54,16 @@ The event ping contains a list of events ([see event format on readthedocs.io](h
...
-- Controller menu button is pressed
-- Controller back button is pressed
-
-e.g. we don't log an event when the overlay is closed automatically after "Home" is selected.
+(**)YouTube tile clicks are a sub-category of bundled tile clicks- each click will count towards both values.
```
Comment (liuche) [inline docs/telemetry.md]: One more nit: (**) doesn't show up properly in the docs in markup (see the docs/Telemetry.md) - even though it shows up fine in the comments. I think the fix is to wrap it in single quotes so: (`**`) (you can look at the raw text of this by editing it). I tested this by editing the docs/Telemetry.md and looking at preview (even though it's not possible to commit).

**Pull request merged**